android/src/main/java/com/tailscale/ipn/App.java

```diff
@@ -94,6 +103,25 @@ String getHostname() {
 		return manu + " " + model;
 	}
 
+	// get user defined nickname from Settings
+	private String getUserConfiguredDeviceName() {
+		String nameFromSystemBluetooth = Settings.System.getString(getContentResolver(), "bluetooth_name");
+		String nameFromSecureBluetooth = Settings.Secure.getString(getContentResolver(), "bluetooth_name");
+		String nameFromSystemDevice = Settings.Secure.getString(getContentResolver(), "device_name");
+
+		Log.d("com.tailscale.ipn.App", "Device name from System Bluetooth: " + nameFromSystemBluetooth);
+		Log.d("com.tailscale.ipn.App", "Device name from Secure Bluetooth: " + nameFromSecureBluetooth);
+		Log.d("com.tailscale.ipn.App", "Device name from System Device: " + nameFromSystemDevice);
+
+		if (notEmpty(nameFromSystemBluetooth)) return nameFromSystemBluetooth;
+		if (notEmpty(nameFromSecureBluetooth)) return nameFromSecureBluetooth;
+		return nameFromSystemDevice;
```

Comment:
You check all the rest for being either null or length 0, but not this one. Which means the func overall can't make any promises on what it does on not finding a match.

I'd use the same `!emptyStr` check for all three, then return `null` on failure to match, and document that on `getUserConfiguredDeviceName`